sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/util/IdentityUtil.java

```diff
@@ -37,11 +47,39 @@ public static String resolveTenantId(String currentTenantId, TokenRequestContext
                     + "`adfs` tenant id and it cannot be replaced with a tenant id challenge provided via "
                     + "TokenRequestContext class. ", null));
             }
-            return CoreUtils.isNullOrEmpty(contextTenantId) ? currentTenantId
+            String resolvedTenantId =  CoreUtils.isNullOrEmpty(contextTenantId) ? currentTenantId
                 : contextTenantId;
+
+            if (resolvedTenantId != currentTenantId && !options.getAdditionallyAllowedTenants().contains(ALL_TENANTS)
+                && !options.getAdditionallyAllowedTenants().contains(resolvedTenantId)) {
+                throw LOGGER.logExceptionAsError(new ClientAuthenticationException("The current credential is not configured to acquire tokens for tenant "
+                    +  resolvedTenantId + ". To enable acquiring tokens for this tenant add it to the AdditionallyAllowedTenants on the credential options, "
+                    + "or add \"*\" to AdditionallyAllowedTenants to allow acquiring tokens for any tenant.", null));
+            }
+            return resolvedTenantId;
         }
 
         return currentTenantId;
 
     }
+
+    public static List<String> resolveAdditionalTenants(List<String> additionallyAllowedTenants) {
+        if (additionallyAllowedTenants == null) {
+            return Collections.emptyList();
+        }
+
+        if (additionallyAllowedTenants.contains(ALL_TENANTS)) {
+            return Arrays.asList(ALL_TENANTS);
+        }
```

Comment:
nit: if the customer specifies `*` as well as other tenets we're going to just delete the other tenets. It seems more reasonable to me to prevent this case in `IdentityClientOptions.setAdditionallyAllowedTenets` (and then saves us reallocating a list here?)